### PR TITLE
Various additions

### PR DESCRIPTION
**Pull request by ararslan**
This PR is comprised of three unrelated commits. The first adds highlighting for the type `AbstractRange`. The second only highlights `in` and `isa` when they're used in an infix position. (This was a pet peeve of mine.) The last adds highlighting for the new Unicode constant to replace the exported `e`.

Example:
screen shot 2017-09-29 at 12 37 05 pm

### CODE REVIEW

**Comment by ararslan** [comment]
Come to think of it, I need to adjust the infix keyword recognition to be version-specific.

Also I noticed that the `ℯ` isn't getting highlighted; could you provide any insight on that, @carlobaldassi?

**Comment by ararslan** [comment]
I changed the infix keyword logic a bit, now this is what it looks like:
screen shot 2017-09-29 at 10 26 32 pm

File: syntax/julia.vim [hunk 1]
```diff
@@ -220,7 +223,7 @@ syntax match   juliaBaseTypeTime	display "\<\%(Date\%(Time\)\?\)\>"
 syntax match   juliaBaseTypeTime0607	display "\<DateFormat\>"
 syntax match   juliaBaseTypeOther	display "\<\%(RemoteRef\|Task\|Condition\|VersionNumber\|IPv[46]\|SerializationState\|WorkerConfig\|Future\|RemoteChannel\|IPAddr\|Stack\%(Trace\|Frame\)\|\(Caching\|Worker\)Pool\|AbstractSerializer\)\>"
 
-syntax match   juliaConstNum		display "\<\%(NaN\%(16\|32\|64\)\?\|Inf\%(16\|32\|64\)\?\|eu\?\|pi\|π\|eulergamma\|γ\|catalan\|φ\|golden\)\>"
+syntax match   juliaConstNum		display "\<\%(NaN\%(16\|32\|64\)\?\|Inf\%(16\|32\|64\)\?\|eu\?\|ℯ\|pi\|π\|eulergamma\|γ\|catalan\|φ\|golden\)\>"
```
Comment (carlobaldassi) [inline syntax/julia.vim]: Most of these are actually deprecated in 0.7, so there should be some versioning going on with `juliaConstNum`.

File: syntax/julia.vim [hunk 2]
```diff
@@ -206,7 +209,7 @@ syntax match   juliaBaseTypeSet0607	display "\<AbstractSet\>"
 syntax match   juliaBaseTypeIO		display "\<\%(IO\%(Stream\|Buffer\|Context\)\?\|RawFD\|StatStruct\|DevNull\|FileMonitor\|PollingFileWatcher\|Timer\|Base64\%(Decode\|Encode\)Pipe\|\%(UDP\|TCP\)Socket\|\%(Abstract\)\?Channel\|BufferStream\|ReentrantLock\)\>"
 syntax match   juliaBaseTypeProcess	display "\<\%(ProcessGroup\|Pipe\|Cmd\)\>"
 syntax match   juliaBaseTypeProcess0607	display "\<PipeBuffer\>"
-syntax match   juliaBaseTypeRange	display "\<\%(Dims\|Range\%(Index\)\?\|\%(Ordinal\|Step\|\%(Abstract\)\?Unit\)Range\|Colon\)\>"
+syntax match   juliaBaseTypeRange	display "\<\%(Dims\|Range\%(Index\)\?\|\%(Ordinal\|Step\|\%(Abstract\)\?\%(Unit\)\?\)Range\|Colon\)\>"
```
Comment (carlobaldassi) [inline syntax/julia.vim]: The new `AbstractRange` should only get highlighted with julia 0.7.
We need a new `juliaBaseTypeRange07` category...

**Comment by carlobaldassi** [comment]
Thanks. I left some minor comments. I'll look into the highlighting of `ℯ`.

**Comment by carlobaldassi** [comment]
Unfortunately, it seems that vim does not recognize `ℯ` as a valid identifier character, which means that the beginning-of-word `\<` and end-of-word `\>` patterns disrupt the highlighting. This makes the implementation of proper highlighting of `ℯ` quite annoying. For the time being, I suggest that we just leave it without the word-delimiters, which means that it will be mistakenly highlighted in the middle of words.

**Comment by ararslan** [comment]
Okay, I've reworked things a bit. In particular (in commit order):

1. I introduced the `juliaBaseTypeRange07` category and it seems to work, but surely it could be done better. I've also not dealt with the `Range` deprecation in 0.7.

2. No parens around `\S`, `%` before parens.

3. I removed this commit and redid it from scratch using the included Julia programs, and I went with your suggestion to highlight script e all the time, even in the middle of words.

**Comment by ararslan** [comment]
Okay, so I take it things are looking okay now 😆

**Comment by carlobaldassi** [comment]
Yep, thanks. I'll take a look at the euler constant thing. (I also have an unrelated commit that I wanted to push.)

**Pull request merged**